cypress/e2e/settings.cy.js

```diff
@@ -1,32 +1,85 @@
 /// <reference types="cypress" />
 /// <reference types="../support" />
 
+import HomePageObject from '../support/pages/home.pageObject';
+import SettingsPageObject from '../support/pages/settings.pageObject';
+import SignInPageObject from '../support/pages/signIn.pageObject';
+
+const faker = require('faker');
+const settingsPage = new SettingsPageObject();
+const homePage = new HomePageObject();
+const signInPage = new SignInPageObject();
+
 describe('Settings page', () => {
-  before(() => {
+  let user;
+
+  const userChanged = {
+    username: faker.name.firstName(),
+    bio: faker.lorem.words(),
+    email: faker.internet.email(),
+    password: faker.internet.password(),
+  };
 
+  const successUpdate = 'Update successful!';
+
+  before(() => {
+    cy.task('generateUser').then((generateUser) => {
+      user = generateUser;
+    });
   });
 
   beforeEach(() => {
-
+    cy.task('db:clear');
+    homePage.visit();
+    cy.login(user.email, user.username, user.password);
   });
 
   it('should provide an ability to update username', () => {
+    settingsPage.visit();
+    settingsPage.updateUsernameField(userChanged.username);
+    settingsPage.clickUpdateSetBtn();
+    settingsPage.assertSuccessfulUpdate(successUpdate);
```

Comment:
The 'assertSuccessfulUpdate' function should ideally take the expected value as a parameter instead of a constant. This way, you make your test more flexible and adaptable to changes in the future.